lib/timers.js

```diff
@@ -232,6 +231,12 @@ function listOnTimeout() {
   } else if (list === refedLists[msecs]) {
     delete refedLists[msecs];
```

Comment:
Would it still work if `this.close()` was placed into both of these `if`cases? I think that may be a better and more robust patch.